File: 0547-number-of-provinces/0547-number-of-provinces.c
void DFS(int i, int **Adj, int* visited, int isConnectedSize) {
    visited[i] = 1; 
    for (int j = 0; j < isConnectedSize; j++) {
        if (Adj[i][j] == 1 && visited[j] == 0) {
            DFS(j, Adj, visited, isConnectedSize);
        }
    }
}

int findCircleNum(int** isConnected, int isConnectedSize, int* isConnectedColSize) {
    int count = 0;
    int visited[isConnectedSize];
    for (int i=0;i<isConnectedSize;i++)
    {
        visited[i] = 0;
    }
    for(int i=0;i<isConnectedSize;i++)
    {
        if(visited[i]==0)
        {
            count ++;
            DFS(i, isConnected, visited, isConnectedSize);
        }
    }
    return count;
    
}

File: 0547-number-of-provinces/0547-number-of-provinces.py
class Solution(object):
    def helper(self, adj_list, visited, node):
        visited.add(node)
        for neighbours in adj_list[node]:
            if neighbours not in visited:
                self.helper(adj_list, visited, neighbours)
        
    
    def findCircleNum(self, isConnected):
        V = len(isConnected)
        
        adj_list = [[] for _ in range(V)]
        
        for i in range(V):
            for j in range(V):
                if isConnected[i][j] == 1 and i != j:
                    adj_list[i].append(j)
                    adj_list[j].append(i)
        

        visited = set()
        count = 0

        for i in range(V):
            if i not in visited:
                self.helper(adj_list, visited, i)
                count += 1
    
        return count

        